### PR TITLE
Issue #14: Added Echo Request Handler for adding Meds.

### PR DESCRIPTION
**Pull request by jking31cs**
Added a simple MedService/MockMedService that adds a medicine by its
name.

Successfully grabbed access token from user object passed into the
Session object for Alexa, and grabbed user information from it.

Added MockAuthUserService that validates against google with the
access token and creates a fake AuthUser with that information.

Moved all Echo specific code into its own package, added postman
tests, and testing the rest endpoints to ensure that they work.

This does successfully use OAuth for the Echo, but
not for the Rest Endpoints.

### CODE REVIEW

File: server/src/test/java/info/doseamigos/meds/MedTest.java [hunk 1]
```diff
@@ -56,10 +56,10 @@ public void testGetName() throws Exception {
         assertEquals(med.getName(), "Test Drug");
     }
 
-    @Test(expectedExceptions = NullPointerException.class)
-    public void testSetName() throws Exception {
-        new Med().setName(null);
-    }
+//    @Test(expectedExceptions = NullPointerException.class)
```
Comment (jking31cs) [inline server/src/test/java/info/doseamigos/meds/MedTest.java]: There was a weird issue where Alexa was doing the wrong intent, and kept causing NPE's, so I made this nullable and commented out the test.  I figured out that the issue was that a trailing space in testing caused it to think it was a different intent.  I might be able to undo this tomorrow once we're in a better more stable state.

File: server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Mock an auth user based on google stuff.
+ */
+public class MockAuthUserService implements AuthUserService {
+
+    private static final Logger log = LoggerFactory.getLogger(MockAuthUserService.class);
+    @Override
```
Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java:23]: Need a newline before this for consistency.

File: server/src/main/java/info/doseamigos/authusers/DefaultAuthUserService.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Implementation of AuthUserService using Google OAuth URLs.
+ */
+public class DefaultAuthUserService implements AuthUserService {
```
Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/DefaultAuthUserService.java:22]: The code works, but only locally.  With Amazon Lambda, it cannot connect to the DB.  I've had all sorts of headaches about this, and this isn't being used for the demo tomorrow.

File: server/src/main/java/info/doseamigos/authusers/AuthUserGuiceModule.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package info.doseamigos.authusers;
+
+import com.google.inject.AbstractModule;
+
+/**
+ * Created by jking31cs on 6/27/16.
```
Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/AuthUserGuiceModule.java]: damnit, need to remove default comments :(

File: server/src/main/java/info/doseamigos/authusers/AuthUser.java [hunk 5]
```diff
@@ -5,22 +5,25 @@
...
+
 import static java.util.Objects.requireNonNull;
 
 /**
  * POJO for users that authenticate with Google.
  */
 public class AuthUser {
 
-    private long authUserId;
+    private Long authUserId;
```
Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/AuthUser.java:15]: Using a Long for new AuthUser creation.

File: server/src/main/java/info/doseamigos/authusers/AuthUser.java [hunk 6]
```diff
@@ -5,22 +5,25 @@
...
     private AmigoUser amigoUser;
     private String email;
+    private BigInteger googleRef;
```
Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/AuthUser.java]: Google's id is too big for a long, so I'm using BigInteger, and for the DB, I'm storing the string representation.

Comment (mckoon) [inline server/src/main/java/info/doseamigos/authusers/AuthUser.java]: Is it ever used as a number?  If not, why not just a String always?

Comment (jking31cs) [inline server/src/main/java/info/doseamigos/authusers/AuthUser.java]: ... good point

**Comment by jking31cs** [comment]
Made the fixes in a separate repo since the demo releases are from the previous commit.  I didn't want to clobber that commit for that reason.

**Comment by mckoon** [comment]
> Made the fixes in a separate repo

You mean "commits"?

**Comment by jking31cs** [comment]
I did mean commits, I'm tired...

File: server/src/main/java/info/doseamigos/authusers/AuthUser.java [hunk 7]
```diff
@@ -5,22 +5,25 @@
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
+import java.math.BigInteger;
```
Comment (mckoon) [inline server/src/main/java/info/doseamigos/authusers/AuthUser.java]: Unused?

File: server/src/main/java/info/doseamigos/authusers/AuthUserDao.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package info.doseamigos.authusers;
+
+import java.math.BigInteger;
```
Comment (mckoon) [inline server/src/main/java/info/doseamigos/authusers/AuthUserDao.java]: Unused?

File: server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
+package info.doseamigos.authusers;
+
+import java.io.IOException;
+import java.math.BigInteger;
```
Comment (mckoon) [inline server/src/main/java/info/doseamigos/authusers/MockAuthUserService.java]: Unused?

**Comment by jking31cs** [comment]
Cleaned up imports

**Comment by mckoon** [comment]
+1

**Pull request merged**